bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,542 @@
+    }
+
+    @NonNullByDefault
+    private class KeepAlive extends Thread {
+
+        @Override
+        public void run() {
+            try {
+                while (!isInterrupted()) {
+                    try {
+                        CGateSession session = cGateSession;
+                        if (session == null || !session.isConnected()) {
+                            if (!getThing().getStatus().equals(ThingStatus.ONLINE))
+                                connect();
+                            else {
+                                updateStatus();
+                            }
+                        } 
+                    } catch (Exception e) {
+                    }
+
+                    sleep(10000l);
+                }
+            } catch (InterruptedException e) {
+            }
+        }
+    }
+
+    private void connect() {
+        CGateSession cGateSession = this.cGateSession;
+        if (cGateSession == null) {
+            cGateSession = CGateInterface.connect(this.ipAddress, 20023, 20024, 20025, new CBusThreadPool());
+            cGateSession.registerEventCallback(new EventMonitor());
+            cGateSession.registerStatusChangeCallback(new StatusChangeMonitor());
+            this.cGateSession = cGateSession;
+        }
+        if (cGateSession.isConnected()) {
+            logger.debug("CGate session reports online");
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            try {
+                cGateSession.connect();
+                updateStatus();
+                if (getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                    initializeChildThings();
+                }
+            } catch (CGateConnectException e) {
+                updateStatus();
+                if (e.getMessage().startsWith("Connection refused")) {
+                    logger.error("Failed to connect to CGate: Connection refused");
+                } else {
+                    logger.error("Failed to connect to CGate: {}", e.getMessage());
+                }
+                try {
+                    cGateSession.close();
+                } catch (CGateException e2) {
+                }
+            }
+        }
+    }
+
+    public void updateStatus() {
+        ThingStatus lastStatus = getThing().getStatus();
+        CGateSession cGateSession = this.cGateSession;
+        if (cGateSession == null)
+            return;
+        if (cGateSession.isConnected()) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            if (!lastStatus.equals(ThingStatus.OFFLINE)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+        }
+        if (!getThing().getStatus().equals(lastStatus)) {
+            boolean isOnline = getThing().getStatus().equals(ThingStatus.ONLINE);
+            updateChildThings(isOnline);
+        }
+    }
+
+    private void initializeChildThings() {
+        threadPool.execute(new Runnable() {
```

Comment:
Please use lambdas instead of an anonymous Runnable

Reply:
Changed both Runnable's to lambdas